### PR TITLE
#14 Get file operation

### PR DESCRIPTION
**Pull request by longtimeago**

### CODE REVIEW

**Comment by longtimeago** [comment]
@alex-palevsky please, find someone for review

**Comment by karato** [comment]
Let me find a reviewer for this pull request, thanks for submitting it

**Comment by karato** [comment]
@alevohin review it

**Comment by ggajos** [comment]
@longtimeago very like your suggestion about path, thanks! Few comments above.

**Comment by longtimeago** [comment]
@ggajos thanks for remarks. Please, take a look again

File: src/test/java/com/nerodesk/LaunchTest.java [hunk 1]
```diff
@@ -88,6 +102,55 @@ public Void call() throws Exception {
     }
 
     /**
+     * Application can return file content.
+     * @throws Exception If fails
+     */
+    @Test
+    @SuppressWarnings("PMD.DoNotUseThreads")
+    public void getFile() throws Exception {
```
Comment (alevohin) [inline src/test/java/com/nerodesk/LaunchTest.java]: Let's rename it to `returnsFileContent`

File: src/test/java/com/nerodesk/TkGetFileTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+     */
+    @Rule
+    public final transient TemporaryFolder temp = new TemporaryFolder();
+
+    /**
+     * TkGetFile can return file content.
+     * @throws Exception If something went wrong
+     */
+    @Test
+    public void returnFileContent() throws Exception {
```
Comment (alevohin) [inline src/test/java/com/nerodesk/TkGetFileTest.java]: Let's rename it to `returnsFileContent`

**Comment by alevohin** [comment]
@longtimeago thank you, looks great! Please see two minor comments above.

**Comment by longtimeago** [comment]
@alevohin Here they are :)

**Comment by alevohin** [comment]
@rultor merge please

File: src/main/java/com/nerodesk/Launch.java [hunk 3]
```diff
@@ -61,9 +78,22 @@
...
+     * @param store Store to start with.
+     * @throws IOException If something goes wrong.
+     */
+    public Launch(final int port, final Storage store) throws IOException {
         Logger.info(Launch.class, "HTTP server starting on port %d", port);
         new FtBasic(
-            new TsRegex().with("/", new TkIndex()),
+            new TsFork(
+                new FkRegex("/", new TkIndex()),
+                new FkRegex(TkGetFile.PATH, new TkGetFile(store))
```
Comment (ggajos) [inline src/main/java/com/nerodesk/Launch.java:100]: Let's inline path, it's used on this level. TkGetFile shouldn't be aware of mapping.

**Comment by ggajos** [comment]
@longtimeago one remark

File: src/main/java/com/nerodesk/Launch.java (hunk 3 above)
Comment (longtimeago) [inline src/main/java/com/nerodesk/Launch.java:100]: We need to use this path in `TkGetFileTest`, so it can't be inlined (otherwise it would be duplicated)

**Comment by ggajos** [comment]
@rultor merge

**Comment by longtimeago** [comment]
@rultor please, merge again

**Comment by longtimeago** [comment]
@ggajos updated puzzles

**Comment by ggajos** [comment]
@rultor merge

**Comment by karato** [comment]
@alevohin paid, thanks, added **17 mins** to your account, payment `53314991`; you're getting extra minutes here (c=2); +17 added to your rating, current score is: [+1428](http://www.netbout.com/b/35105?open=rating)

**Comment by karato** [comment]
@rultor deploy now pls

**Pull request merged**